ddl/generated_column.go

```diff
@@ -228,6 +228,7 @@ func checkModifyGeneratedColumn(tbl table.Table, oldCol, newCol *table.Column, n
 type illegalFunctionChecker struct {
 	hasIllegalFunc bool
 	hasAggFunc     bool
+	hasRowVal      bool // hasRowVal checks whether the functional index refers to a row value
```

Comment:
Note: generated column doesn't allow row function too.

Reply:
OK, thx. Will the error message of generated column be the same as generated index?

Reply:
No， you can check it yourself.

Reply:
done

PS: add an errno according to MySQL 8.0. Where should I sync the added errno? 'Cause I noticed that there are also errno list in `pingcap/parser` repo; I'm not sure if the sync is needed, and if there is other repo with the errno list.